bytes.go

```diff
@@ -65,19 +70,47 @@ func logn(n, b float64) float64 {
 	return math.Log(n) / math.Log(b)
 }
 
-func humanateBytes(s uint64, base float64, sizes []string) string {
+func precisionSprint(val float64, precision int) string {
+	_, frac := math.Modf(val)
+	if frac == 0 {
+		return fmt.Sprintf("%.0f", val)
+	}
+	f := fmt.Sprintf("%%.%df", precision)
+	res := fmt.Sprintf(f, val)
```

Comment:
FYI: 
you can simplify these two lines to:
```go
res := fmt.Sprintf("%.*f",precision,val)
```
[play](https://play.golang.org/p/Ofg9q-ib-Av)